cmd/zc_enumerator.go

```diff
@@ -157,14 +157,20 @@ func initResourceTraverser(source string, location common.Location, ctx *context
 
 	// Feed list of files channel into new list traverser, separate SAS.
 	if listofFilesChannel != nil {
-		splitsrc := strings.Split(source, "?")
 		sas := ""
+		if location.IsRemote() {
+			srcURL, err := url.Parse(source)
```

Comment:
I noticed that this is different from how we remove the SAS's in copy.go (about lines 800 to 950).  There, we call azblob.NewBlobURLParts (or the corresponding Files or BlobFS functions).  Internally, they do something subtly different from what's here: the removed only those query parameters that actually are part of a SAS, leaving all others in place.

I'm not sure how often folks will give us additional query params.  Probably not very often.  I can imagine someone trying to use the include= parameter when listing blobs, to get deleted blobs, snapshots or whatever, and I can imagine in a single-blob scenario someone might use snaphshot=...  Although I'm not sure any of those would actually work, because I doubt we test them.

@zezha-msft What do you think?  Is it important that we always remove SAS's with the strictly correct method that does not touch any other unrelated parts of the query string that may exist?  Or, should we perhaps leave this code here as is, and in fact return errors back in copy.go if there are any additional query parameters left after we remove the SAS?  That would at least prevent users from trying stuff where the results are undefined because we don't test them.  I suspect the latter option would be quickest and safe to implement.

Reply:
I think we should have a single function to do this and unify the SAS splitting.

Reply:
We do it in a lot of places, and it tends to cause some code-duplication-creep.

Reply:
Resolved.